python/cudf/cudf/core/multiindex.py

```diff
@@ -1007,10 +1007,15 @@ def _concat(cls, objs):
 
         source_data = cudf.DataFrame._concat(source_data)
         names = [None] * source_data._num_columns
-        objs = list(filter(lambda o: o.names is not None, objs))
-        for o in range(len(objs)):
-            for i, name in enumerate(objs[o].names):
-                names[i] = names[i] or name
+        objs = set(
+            map(lambda x: x.names, filter(lambda o: o.names is not None, objs))
+        )
+        if len(objs) == 1:
+            # Only set names if all `objs` have same `names`,
+            # else ignore.
+            for o in objs:
+                for i, name in enumerate(o):
+                    names[i] = names[i] or name
```

Comment:
This logic is kind of hard to follow, how about:
```python
try:
    # Only set names if all objs have the same names
    (names,) = {o.names for o in obj} - {None}
except ValueError:
   names = [None] * source_data._num_columns
```

Note that names is initialised to be a list of `None` on line 1009, so the `names[i] or name` check only ever produces `name`.

Reply:
Done